sources/appendix-value-extraction.asciidoc

```diff
@@ -0,0 +1,437 @@
+====
+
+==== Default extractors
+
+Compatible implementations provide extractors for the following types out of the box:
+
+* arrays of objects and all primitive data types
+* `java.util.Iterable` (when `@Valid` is given on the iterable element itself, the element and all its entries will be validated; this is to grant backwards compatability with BV 1.1; when `@Valid` is given on the type parameter of an iterable element, all the entries will be validated). A node of type `TYPE_PARAMETER` and with the name "<collection element>" will be added to the path when validating a type parameter constraint.
+* `java.util.Map` (when `@Valid` is given on the map element itself, the element and all its values will be validated; this is to grant backwards compatability with BV 1.1; when `@Valid` is given on the key type parameter, only the keys will be validated; when `@Valid` is given on the value type parameter, only the values will be validated. A node of type `TYPE_PARAMETER` and with the name "<map value>" will be added to the path when validating a type parameter constraint on the map value. A node of type `TYPE_PARAMETER` and with the name "<map key>" will be added to the path when validating a type parameter constraint on the map key.
+TODO: format of key representations
+* `java.util.Optional`. No node will be appended to the path when validating type-level constraints on `Optional`. Constraints given on an element of type `Optional` apply to the wrapped value by default (see below).
+* `javafx.beans.observable.ObservableValue`. No node will be appended to the path when validating type-level constraints on `ObservableValue`. Constraints given on an element of type `ObservableValue` apply to the wrapped value by default (see below).
+
+==== Plugging in custom extractors
+
+Additional value extractors can be registered with the Bean Validation provider in the following ways:
+
+* Invoke the new method `Configuration#addValueExtractor(ValueExtractor<?>)` (to apply it on the validation factory level)
+* Invoke the new method `ValidatorContext#addValueExtractor(ValueExtractor<?>)` (to apply it for a single `Validator`)
+* Specify the fully qualified class name of one or several extractors in `META-INF/validation.xml`: 
+
+[source,xml]
+----
+<value-extractors>
+    <value-extractor>com.example.MyExtractor</value-extractor>
+</value-extractors>
+----
+
+==== Retrieval of extractors
+
+When detecting a type-level constraint or cascade, the applicable extractor is determined as follows:
+
+* Choose the most specific extractor matching the type parameter in question; an extractor A is more specific than another extractor B if A extracts a subtype of the type extracted by B.
```

Comment:
The reasons I ask is that the sentence made me think you were mentioning the the type you extract, not the container type. because you mentioned "the type parameter in question"